### PR TITLE
Add support for skew animations

### LINKED ISSUE #1915: New rendering engine ignores Skew parameter
## Which Version of Lottie are you using?

Lottie 4.0.1

## Expected Behavior 

https://user-images.githubusercontent.com/20406383/212380586-dbc1be60-75e7-40e5-8a88-f1e82b1af8e9.MP4

## Actual Behavior

https://user-images.githubusercontent.com/20406383/212380763-f8e5da23-3053-4e47-a01b-be9ec19da1e5.MP4

## Animation JSON

[AnimationWithSkew.json.zip](https://github.com/airbnb/lottie-ios/files/10413865/AnimationWithSkew.json.zip)

### PR DESCRIPTION
**Pull request by calda**
This PR adds support for skew animations. Fixes #1915.

| Before | After |
| ----- | ----- |
| 1915 before | 1915 after |

### CODE REVIEW

File: Sources/Private/Model/ShapeItems/ShapeTransform.swift [hunk 1]
```diff
@@ -115,6 +115,15 @@ final class ShapeTransform: ShapeItem {
   /// Skew Axis
   let skewAxis: KeyframeGroup<LottieVector1D>
 
+  /// Whether or not this transform has a non-zero skew value
+  var hasSkew: Bool {
+    guard !skew.keyframes.isEmpty, !skewAxis.keyframes.isEmpty else {
+      return false
+    }
+
+    return skew.keyframes.contains(where: { $0.value.cgFloatValue != 0 })
```
Comment (campersau) [inline Sources/Private/Model/ShapeItems/ShapeTransform.swift:124]: Should `skewAxis.keyframes` also be checked with contains here?

Comment (calda) [reply Sources/Private/Model/ShapeItems/ShapeTransform.swift:124]: No -- the sample animation added in this PR has a skew of 30 and a skew axis of 0. So having a non-zero axis is not a requirement for rendering a skew.

File: Sources/Private/Utility/Primitives/VectorsExtensions.swift [hunk 2]
```diff
@@ -339,3 +339,26 @@ extension CATransform3D {
...
+      && lhs.m23 == rhs.m23
+      && lhs.m24 == rhs.m24
+      && lhs.m31 == rhs.m31
+      && lhs.m32 == rhs.m32
+      && lhs.m33 == rhs.m33
+      && lhs.m34 == rhs.m34
+      && lhs.m41 == rhs.m41
+      && lhs.m42 == rhs.m42
+      && lhs.m43 == rhs.m43
+      && lhs.m44 == rhs.m44
```
Comment (erichoracek) [inline Sources/Private/Utility/Primitives/VectorsExtensions.swift:362]: Can we use `CATransform3DEqualToTransform` instead?

File: Sources/Private/Utility/Primitives/VectorsExtensions.swift [hunk 3]
```diff
@@ -339,3 +339,26 @@ extension CATransform3D {
     CATransform3DConcat(CATransform3D.makeSkew(skew: skew, skewAxis: skewAxis), self)
   }
 }
+
+// MARK: - CATransform3D + Equatable
+
+extension CATransform3D: Equatable {
```
Comment (erichoracek) [inline Sources/Private/Utility/Primitives/VectorsExtensions.swift:345]: If someone else has this conformance will this definition conflict? Wondering if we could avoid this if so

**Review by erichoracek — APPROVED** [review]

File: Sources/Private/Utility/Primitives/VectorsExtensions.swift (hunk 3 above)
Comment (calda) [reply Sources/Private/Utility/Primitives/VectorsExtensions.swift:345]: LayerProperty has an Equatable constraint right now. Good point about this conformance potentially causing conflicts. I'll look at making the Equatable constraint optional.

File: Sources/Private/Utility/Primitives/VectorsExtensions.swift (hunk 2 above)
Comment (calda) [reply Sources/Private/Utility/Primitives/VectorsExtensions.swift:362]: Aha thanks! Not used to the naming schemes of these helpers so I didn't find that

File: Sources/Private/Utility/Primitives/VectorsExtensions.swift (hunk 3 above)
Comment (calda) [reply Sources/Private/Utility/Primitives/VectorsExtensions.swift]: https://github.com/airbnb/lottie-ios/pull/1917/commits/82ffdf284a4990928d98ff8ac799c34e5fcb20b0

**Pull request merged**